kaldb/src/main/java/com/slack/kaldb/chunkManager/ChunkManagerBase.java

```diff
@@ -191,4 +194,11 @@ private SearchResult<T> incrementNodeCount(SearchResult<T> searchResult) {
   public List<Chunk<T>> getChunkList() {
     return chunkList;
   }
+
+  @Override
+  public Map<String, FieldType> getSchema() {
+    Map<String, FieldType> schema = new HashMap<>();
+    chunkList.forEach(chunk -> schema.putAll(chunk.getSchema()));
+    return Collections.unmodifiableMap(schema);
+  }
```

Comment:
This is a good observation, and will indeed overwrite the values provided by other chunks. In this case the last-in "wins". In a properly running cluster we would not expect chunks to have different schemas for the same field, as that would mean someone is producing mixed content for the same field.

We haven't fully resolved what to do in this case - the idea is that for each indexer it would index it according to the predefined schema or for a newly emergent field the first type it sees. Conflicts from then on out would not be indexed to that field would would index to another field (ie, `_field_conflicts`). Since this is not a distributed decision however, it's possible that different indexer would have different types for this field depending on which version they saw first.

When we query for the schema of the cluster we now have an issue - what do we do when different nodes have different opinions about schema? This PR leaves this as "undefined" behavior as we haven't fully finished the schema work.